feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -1,29 +1,121 @@
+import {
+  AssociatedObject,
+  Category,
+  TicketStarter,
+} from '@spartacus/customer-ticketing/root';
 import { FormUtils } from '@spartacus/storefront';
-import { Observable } from 'rxjs';
+import { Observable, Subscription } from 'rxjs';
 import { CustomerTicketingDialogComponent } from '../../shared/customer-ticketing-dialog/customer-ticketing-dialog.component';
 @Component({
   selector: 'cx-customer-ticketing-create-dialog',
   templateUrl: './customer-ticketing-create-dialog.component.html',
 })
 export class CustomerTicketingCreateDialogComponent
   extends CustomerTicketingDialogComponent
-  implements OnInit
+  implements OnInit, OnDestroy
 {
   ticketCategories$: Observable<Category[]> =
     this.customerTicketingFacade.getTicketCategories();
   ticketAssociatedObjects$: Observable<AssociatedObject[]> =
     this.customerTicketingFacade.getTicketAssociatedObjects();
+  subscription: Subscription;
+  ticketStarter: TicketStarter;
+  ticketCategories: Array<Category>;
+  ticketAssociatedObjects: Array<AssociatedObject>;
+
+  @Input()
+  selectedCategory: Category;
+
+  @Input()
+  selectedAssociatedObject: AssociatedObject;
+
+  protected getCreateTicketPayload(form: FormGroup): TicketStarter {
+    return {
+      message: form?.get('message')?.value,
+      subject: form?.get('subject')?.value,
+      associatedTo: this.selectedAssociatedObject,
+      ticketCategory: this.selectedCategory,
+    };
+  }
 
   ngOnInit(): void {
     this.buildForm();
   }
 
+  protected buildForm(): void {
+    const form = new FormGroup({});
+    form.setControl(
+      'subject',
+      new FormControl('', [
+        Validators.required,
+        Validators.maxLength(this.inputCharactersForSubject),
+      ])
+    );
+    form.setControl(
+      'ticketCategory',
+      new FormControl('', [Validators.required])
+    );
+    form.setControl('associatedTo', new FormControl());
+    form.setControl(
+      'message',
+      new FormControl('', [
+        Validators.required,
+        Validators.maxLength(this.inputCharactersLimit),
+      ])
+    );
+    form.setControl(
+      'file',
+      new FormControl('', [
+        this.filesFormValidators.maxSize(this.maxSize),
+        this.filesFormValidators.maxEntries(this.maxEntries),
+      ])
+    );
+    this.form = form;
+  }
+
+  setSelectedAssociatedObject(selectedAssociatedObjectIndex: number) {
+    this.ticketAssociatedObjects$.subscribe((associatedObjects) => {
+      this.ticketAssociatedObjects = associatedObjects;
+    });
+    this.selectedAssociatedObject =
+      this.ticketAssociatedObjects[selectedAssociatedObjectIndex - 1];
+    this.form.controls.associatedTo.setValue(
+      this.selectedAssociatedObject.code
+    );
+  }
+
+  setSelectedCategory(selectedCategoryIndex: number): void {
+    this.ticketCategories$.subscribe((categories) => {
+      this.ticketCategories = categories;
+    });
+
+    this.selectedCategory = this.ticketCategories[selectedCategoryIndex - 1];
```

Comment:
Is it really a 1-based index to need to subtract one?

Reply:
Yes, selectedindex is 1-based..also this is required since I added another option to show 'Select Category' in the form of placeholder